apps/prairielearn/assets/scripts/question.ts

```diff
@@ -229,6 +292,33 @@ function setupDynamicObjects() {
   }
 }
 
+function loadPendingSubmissionPanel(this: HTMLDivElement) {
+  const { submissionId, dynamicRenderUrl } = this.dataset;
+  if (submissionId == null || dynamicRenderUrl == null) return;
+
+  const wasModalOpen = ($(`#submissionInfoModal-${submissionId}`).data('bs.modal') || {})._isShown;
+  $(`#submissionInfoModal-${submissionId}`).modal('hide');
+
+  fetch(dynamicRenderUrl)
+    .then(async (response) => {
+      const msg = await response.json();
```

Comment:
We should probably check the status here and throw an error here for anything other than a 200 so that we can show the "Error retrieving submission" message.

Reply:
Done in 7d1ff914f9d8a42f893a698427e9faa31c44b9bd. This has been copied from jquery, where `fail()` is called for a non-200 response, which is not the case in `fetch`.